src/modules/dashboard/components/Whitelist/CSVUploader/CSVUploader.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Papa, { ParseResult } from 'papaparse';
+import { useField } from 'formik';
+import { MessageDescriptor } from 'react-intl';
+import isNil from 'lodash/isNil';
+
+import FileUpload, { UploadFile } from '~core/FileUpload';
+import { isAddress } from '~utils/web3';
+
+import CSVUploaderItem from './CSVUploaderItem';
+
+interface Props {
+  name: string;
+  error?: string | MessageDescriptor;
+  processingData: boolean;
+  setProcessingData: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const MIME_TYPES = ['text/csv'];
+
+const CSVUploader = ({
+  name,
+  error,
+  processingData,
+  setProcessingData,
+}: Props) => {
+  const [CSVFile, setCSVFile] = useState<File | null>(null);
+  const [parsedCSV, setParsedCSV] = useState<ParseResult<unknown> | null>(null);
+  const [, { value }, { setValue }] = useField<UploadFile[]>(name);
+
+  const handleUploadError = async () => {
+    if (isNil(value[0].parsedData)) {
+      await setValue([{ ...value[0], parsedData: [] }]);
+    }
+  };
+
+  useEffect(() => {
+    if (CSVFile && !parsedCSV) {
+      Papa.parse(CSVFile, {
```

Comment:
Papa 😆 i feel that not only i have problem with naming, but authors of libs have same problem xD

Reply:
You should check the `FAQ` of this lib, the author really likes the names and it shows on the puns that he shamelessly does